infrastructure/cli/commands/app/index.ts

```diff
@@ -0,0 +1,114 @@
+  if (!deploymentDetails) {
+    throw new Error(`Cannot find deployment ${deployment}`)
+  }
+  const containers = deploymentDetails.spec?.template.spec?.containers || []
+  const currentVersion = containers
+    .map(container => container.image)
+    .filter(image => image && image.startsWith(`${Client.quayUrl}:`))
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    .map(image => image!.slice(Client.quayUrl.length + 1))[0]
+  return currentVersion || null
+}
+
+type ExecScriptOptions = {
+  namespace: string
+  pod: string
+  input: string
+} & Partial<OutputOptions>
+
+export async function execScript(options: ExecScriptOptions & {output: 'stdout'}): Promise<void>
+export async function execScript(options: ExecScriptOptions & {output: 'text'}): Promise<string>
+export async function execScript(options: ExecScriptOptions & {output: 'object'}): Promise<any>
+export async function execScript(options: ExecScriptOptions): Promise<any>
+export async function execScript({
+  namespace,
+  pod,
+  input,
+  output = 'stdout',
+}: ExecScriptOptions): Promise<void | string | any> {
+  process.stderr.write(`Running script on pod "${pod}"…\n`)
+  const kubernetes = new KubernetesApi()
+  return await kubernetes.exec(namespace, pod, ['node', '-'], {input, output})
+}
+
+type RemoteRequestOptions = Partial<OutputOptions>
+type Response<T> = {
+  [pod: string]: T
+}
+
+export async function remoteRequest(
+  environment: Environment,
+  url: string,
+  options: RemoteRequestOptions & {output: 'stdout'}
+): Promise<void>
+export async function remoteRequest(
+  environment: Environment,
+  url: string,
+  options: RemoteRequestOptions & {output: 'text'}
+): Promise<Response<string>>
+export async function remoteRequest(
+  environment: Environment,
+  url: string,
+  options: RemoteRequestOptions & {output: 'object'}
+): Promise<Response<any>>
+export async function remoteRequest(
+  environment: Environment,
+  url: string,
+  options: RemoteRequestOptions
+): Promise<any>
+export async function remoteRequest(
+  environment: Environment,
+  url: string,
+  {output = 'stdout'}: RemoteRequestOptions = {}
+): Promise<Response<string | any> | void> {
+  const input = renderTemplate('scripts/remote/request.njk', {environment, url})
+  const ns = namespace(environment)
+  const kubernetes = new KubernetesApi()
+  const pods = await kubernetes.getPodNames(ns)
+  if (!pods.length) {
+    process.stderr.write('No pods found to run script on.\n')
+  } else {
+    const responses: Response<string | any> = {}
+    for (const pod of pods) {
+      const response = await execScript({namespace: ns, pod, input, output})
+      if (output === 'object' || output === 'text') {
+        responses[pod] = response
+      }
+    }
+    if (output === 'object' || output === 'text') {
+      return responses
+    }
+  }
+}
+
+export const versionRegex = /^(?<buildDate>\d{4}-\d{2}-\d{2})\.(?<ciBuildNumber>\d+)\.(?<commitHash>[0-9a-f]{7})$/i
```

Comment:
`versionRegex` was fairly arbitrarily placed in this file because the image version is a property of the app as a whole… have moved this var to where it's actually used